### DISCUSSION TITLE
Hotbars+ v2.0 is in development

### DISCUSSION THREAD

**Discussion by VideoGameSmash12** [VideoGameSmash12/HotbarsPlus#15] [Announcements]
It has been quite a while since I last updated everyone on what's going on, so I figured I might as well make an announcement about what I've been up to.

Essentially, Hotbars+ is being rewritten for v2.0. While the way the mod works will remain the same, the mod will be structured very differently to allow for multi-version support.

Here's what's planned:
+ **Support for multiple versions of Minecraft (1.14 to 1.18)**
Borrowing the build structure used by ViaFabric, the mod will allow for one jar file to be used across several different versions of Minecraft. The mod will essentially work with versions 1.14 to 1.18 without a sweat.

+ **Hotbar caching**
Once a hotbar page is loaded, it'll be stored into memory for fast hot-swapping.

+ **Yet another absurd increase in page count**
Java's BigInteger will be used for storing the current page in the mod. The reason is elementary: the highest number you can go with it is ***massive***. It's like 600 MB large when stored in a text file. It's basically infinite.

I'll release more details as they become available.

**Comment by JoshuaVandaele** [comment]
I can't help but notice you are not planning to support minecraft 1.6.4, is that a mistake???

**Reply by VideoGameSmash12** [reply]
No.

**Comment by VideoGameSmash12** [comment]
It's been a while, but now I've put up what I've done so far on the `overhaul` branch, which has now been set as the default branch. Assuming I didn't fuck anything up, you'll be able to compile indev versions as I push them.

**Comment by VideoGameSmash12** [comment]
The first prerelease of Hotbars+ v2 has been released. You can download it here:
https://github.com/VideoGameSmash12/HotbarsPlus/releases/tag/v2.0-pre1

**Comment by VideoGameSmash12** [comment]
The sixth prerelease of Hotbars+ v2 has been released. You can download it here:
https://github.com/VideoGameSmash12/HotbarsPlus/releases/tag/v2.0-pre6

**Comment by VideoGameSmash12** [comment]
The ninth prerelease of Hotbars+ v2 has been released. You can download it here:
https://github.com/VideoGameSmash12/HotbarsPlus/releases/tag/v2.0-pre9

You can also download it from both Modrinth and CurseForge:
https://modrinth.com/mod/hotbars%2B
https://beta.curseforge.com/minecraft/mc-mods/hotbarsplus/files/4426661
